SalesforceCommerceCloud/pwa-kit
# Merge Commerce/Einstein Configs into Default Config

# Description



With the addition of "environment specific" configuration support added in a recent update to the SDK it makes sense that we move the API client configurations for scapi/ocapi and einstein into the new default config file. 

In this PR we more the above mentioned configurations as well as fix any tests that this will break as we no longer have `pwa-kit-config.json`. 

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Move commerce api and einstein api configurations into `config/default.js`
- Fix any broken tests (NOTE: I'm directly importing the default config in tests because of the way they are written, accessing them via the `getConfig` utility might mean more refactoring)
- Update the generator to use this new default.js config file with the merged api client configs.
- Removed `config/default.json` as this was an artifact left in from the original PR and shouldn't be there.

# How to Test-Drive This PR

- Test that generating a project works by running: 
```
> GENERATOR_PRESET=test-project node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/generated-project
```
- Smoke test the above project but running it, also manually check that the `config/default.js` file is created and has all the api configuration in it

# Checklists




## General
